core/src/com/softwarearchitecture/networking/persistence/DAO.java

```diff
@@ -1,5 +1,102 @@
 package com.softwarearchitecture.networking.persistence;
 
-public interface DAO {
+import java.util.List;
+
+/**
+ * Represents a Data Access Object (DAO), following the Data Access Object pattern.
+ * This abstract base class abstracts the interaction with the data storage system, allowing for operations
+ * on stored data without exposing the details of the implementation. It provides a standardized
+ * method to access, modify, and manage data across different types of storage mechanisms.
+ * 
+ * @param <T> The type of the object managed by this DAO.
+ * @param <K> The type of the primary key used to identify objects of type T.
+ * 
+ * @see <a href="https://en.wikipedia.org/wiki/Data_access_object">Data Access Object on Wikipedia</a>
+ */
+public abstract class DAO<T, K> {
```

Comment:
Is this the most intuitive way to implement the type before the key? Maybe DAO<K, T>

Reply:
Yes like with how Hashmaps does it. I totally agree